src/main/java/org/zalando/nakadi/security/Client.java

```diff
@@ -1,23 +1,23 @@
 package org.zalando.nakadi.security;
 
-public interface Client {
-
-    Client PERMIT_ALL = clientId -> true;
-
-    boolean is(String clientId);
+import java.util.Set;
 
-    class Authorized implements Client {
-
-        private final String clientId;
+public interface Client {
 
-        public Authorized(final String clientId) {
-            this.clientId = clientId;
+    Client PERMIT_ALL = new Client() {
+        @Override
+        public boolean is(final String clientId) {
+            return true;
         }
 
         @Override
-        public boolean is(final String clientId) {
-            return this.clientId.equals(clientId);
+        public boolean hasNoScopes(final Set<String> scopes) {
+            return false;
         }
-    }
+    };
+
+    boolean is(final String clientId);
+
+    boolean hasNoScopes(final Set<String> scopes);
```

Comment:
Yes, I agree, but in that case in ScopeHelper I will use '!' and that makes code less readable.